h2o-algos/src/main/java/hex/rulefit/RuleFit.java

```diff
@@ -77,6 +78,7 @@ public RuleFit(boolean startup_once) {
     public void init(boolean expensive) {
         super.init(expensive);
         if (expensive) {
+            _parms.validate(this);
```

Comment:
Maybe can the validation be disabled by default and a user could use system property to enable it? This is the way how we provide checks like this in tree algos. Or how expensive is this check in the context of this algo? WDYT @michalkurka ?

Reply:
currently it checks only 1 check: whether min_rule_length is not greater than max_rule_length. Default values are min_rule_length = 3, max_rule_length = 3, so when you set for example max_rule_length = 2 and forget to set also min_rule_length so that is <= 2, rfit will end up with some quite unclear error, so this is to end up with more informative error on what went wrong

Reply:
Ok, make sense to have this check here.